R/qualification-time-profile.R

```diff
@@ -78,144 +81,186 @@ plotQualificationTimeProfiles <- function(configurationPlan, settings) {
+    obsData$yAxis <- ifNotNull(obsData$yAxis, FALSE)
+    plotConfiguration <- updateQualificationTimeProfilePlotConfiguration(
+      simulatedMetaData = simMetaData,
+      observedMetaData = obsMetaData,
+      requestedAxes = requestedAxes,
+      axesProperties = axesProperties,
+      plotConfiguration = plotConfiguration
+    )
+    
+    plotObject <- tlf::plotTimeProfile(
+      data = simData,
+      observedData = obsData,
+      dataMapping = simDataMapping,
+      observedDataMapping = obsDataMapping,
+      plotConfiguration = plotConfiguration
     )
-    return(outputCurve)
+    return(plotObject)
   }
-  # Remove simulation name from path
-  outputPath <- ospsuite::toPathString(pathArray[-1])
-  # Get and convert output path values into display unit
-  simulationQuantity <- ospsuite::getQuantity(outputPath, simulation)
-  simulationPathResults <- ospsuite::getOutputValues(simulationResults, quantitiesOrPaths = outputPath)
-  molWeight <- simulation$molWeightFor(outputPath)
-
-  time <- ospsuite::toUnit(
-    "Time",
-    simulationPathResults$data[, "Time"],
-    axesProperties$x$unit
+  # If all Y, Y2 and Y3 axes are defined and used
+  # Keep Y and Y2 with a warning message
+  if (isOfLength(requestedAxes, 3)) {
+    logError(messages$warningTooManyAxes())
+    requestedAxes <- c("Y", "Y2")
+  }
+  if (isOfLength(requestedAxes, 1)) {
+    axesProperties$y <- axesProperties[[tolower(requestedAxes)]]
+    simData$yAxis <- ifNotNull(simData$yAxis, FALSE)
+    obsData$yAxis <- ifNotNull(obsData$yAxis, FALSE)
+    plotConfiguration <- updateQualificationTimeProfilePlotConfiguration(
+      simulatedMetaData = simMetaData,
+      observedMetaData = obsMetaData,
+      requestedAxes = requestedAxes,
+      axesProperties = axesProperties,
+      plotConfiguration = plotConfiguration
+    )
+    
+    plotObject <- tlf::plotTimeProfile(
+      data = simData,
+      observedData = obsData,
+      dataMapping = simDataMapping,
+      observedDataMapping = obsDataMapping,
+      plotConfiguration = plotConfiguration
+    )
+    plotObject <- updatePlotAxes(plotObject, axesProperties)
+    return(plotObject)
+  }
+  # Y, Y2 and Y3 defined and used
+  if (isOfLength(requestedAxes, 3)) {
+    logError(messages$warningTooManyAxes(
+      project = timeProfilePlan$Project,
+      simulation = timeProfilePlan$Simulation
+    ))
+    requestedAxes <- c("Y", "Y2")
+  }
+  # Use dual axis but needs to map them first
+  # New Y axis will be either Y or Y2 (ie. axis found first because they were sorted)
+  # New Y2 axis will be either Y2 or Y3 (ie. axis found last because they were sorted)
+  axesProperties$y <- axesProperties[[head(tolower(requestedAxes), 1)]]
+  axesProperties$y2 <- axesProperties[[tail(tolower(requestedAxes), 1)]]
+
+  simData$yAxis <- ifNotNull(simData$yAxis, simData$yAxis %in% tail(requestedAxes, 1))
+  obsData$yAxis <- ifNotNull(obsData$yAxis, obsData$yAxis %in% tail(requestedAxes, 1))
```

Comment:
@pchelle Can you explain what happens in these 2 lines?

Reply:
For the right axis feature in the `{tlf}` package, the variable mapping which axis is used needs to provide logical values such as `TRUE` correspond to values on the right axis and `FALSE` values on the left axis.

For intellisense, I am replacing yAxis by y2Axis while adding the explanations below
- `FALSE` y2Axis values correspond to data on the left axis
- `TRUE` y2Axis values correspond to data on the right axis
- Since yAxis values are included in "Y", "Y2" or "Y3", `TRUE` values will correspond to the values included (`%in%`) last found axis